core/src/main/java/jenkins/util/VirtualFile.java

```diff
@@ -168,12 +169,35 @@
     public abstract boolean exists() throws IOException;
 
     /**
-     * Lists children of this directory.
+     * Lists children of this directory. Only one level deep.
```

Comment:
What's the difference between this and listOnlyDescendants then?

Reply:
> Lists only the children that are descendant of the root directory

The important concept is the "root". It's not the current file, it's the file that generates directly or indirectly that current file.

In our case, the root is the workspace or the artifact folder, the current VirtualFile that will call this method, could be 2 layers "inside" that root folder. The `list` will list all the children of the current folder, but the `listOnlyDescandants` will only list the children that are effectively also "inside" the root folder (forbiding the symlink to outside of the workspace).

Reply:
Sorry I didn't explain myself clearly. I understand what you just explained, but my comment was because in the javadoc for the "list" method you added the "one level deep" comment, which is not true for such method IIUC.

Reply:
> which is not true for such method IIUC

Could you elaborate ?

Initially I added the "one level deep" because I investigated the `list()` and `list(...)`, the version with arguments is actually recursive, compared to the no-arg version. I was confused by this naming, so I added the comment.

Reply:
So even for the list() method, it is only one level deep?

Reply:
The `list()` and `listOnlyDescandants()` are ony one level deep but not the `list(...)`